mobile_app/test/test_trigger_build.py

```diff
@@ -0,0 +1,139 @@
+    """
+    Helper to kick off the trigger build task
+    """
+    trigger_build.run_trigger_build(
+        [
+            "--repo-path", repo_path,
+            "--branch-name", TEST_BRANCH
+        ],
+        environ
+    )
+
+
+def _dummy_environ():
+    """
+    Generates a test config for all known keys
+    """
+    result = {}
+    for key in trigger_build.EXPECTED_ENVIRONMENT_VARIABLES:
+        result[key] = "test-value"
+    return result
+
+
+def _make_temp_repo():
+    """
+    Creates a temporary repository for testing
+    """
+
+    repo_path = tempfile.mkdtemp(prefix="test-repo")
+    repo = git.Repo.init(repo_path)
+    dummy_file_path = os.path.join(repo_path, "README")
+
+    # Add a stub file. Git works funny with empty repos
+    open(dummy_file_path, 'wb').close()
+    repo.index.add([dummy_file_path])
+    repo.index.commit("initial commit")
+
+    repo.create_remote('origin', "https://example.com")
+
+    return repo_path
+
+
+class TriggerBuildTestCase(TestCase):
+    """
+    Test cases for the trigger build task
+    """
+
+    @patch('sys.exit', new=Mock(side_effect=Aborted))
+    def test_missing_arguments(self):
+        """
+        Tests that the command quits when not given
+        enough arguments
+        """
+        with self.assertRaises(Aborted):
+            trigger_build.run_trigger_build([], {})
+
+    @patch('git.Remote.pull')
+    def test_branch_already_exists(self, _):
+        """
+        Tests that the task fails if the branch
+        already exists
+        """
+        repo_path = _make_temp_repo()
+        repo = git.Repo(repo_path)
+        repo.create_head(TEST_BRANCH, force=False)
+        environ = _dummy_environ()
+        with self.assertRaises(trigger_build.BranchAlreadyExists):
+            _trigger_build(repo_path, environ)
+
+    def test_missing_env_variable(self):
+        """
+        Test that a missing expected environment variable causes
+        the task to fail.
+        """
+        repo_path = _make_temp_repo()
+
+        with self.assertRaises(trigger_build.MissingEnvironmentVariable):
+            _trigger_build(repo_path, {})
+
+    @patch('git.Remote.pull')
+    @patch('git.Remote.push')
+    def test_config_file_written(self, _, __):
+        """
+        Test that the configuration passed in is properly written to
+        the expected file and is committed.
+        """
+        repo_path = _make_temp_repo()
+        environ = _dummy_environ()
+        _trigger_build(repo_path, environ)
+
+        config_path = os.path.join(repo_path, trigger_build.CONFIG_FILE)
+        config = json.load(file(config_path))
+
+        for key, value in environ.iteritems():
+            self.assertEqual(config[key], value)
+
+    @patch('git.Remote.pull')
+    @patch('git.Remote.push')
+    def test_branch_state(self, _, __):
+        """
+        Verifies that the change has been committed to a new branch
+        with the appropriate name
```

Comment:
Does this docstring and test method name accurately reflect the test?
Isn't the test verifying that the content of the committed file matches the content of the file on-disk?